pkg/client/client.go

```diff
@@ -362,6 +370,11 @@ func (c *Client) Write(ctx context.Context, query string, opts ...WriteOption) (
 		}
 		return res.Transaction.Hash, nil
 	}
+	if c.tblContract == nil {
+		return "", errors.New(
+			"clinet created with no provider, must provide an Infura API key, Alchemy API key, or an ETH backend",
+		)
+	}
```

Comment:
If you expand the code above, you can see this check only happens in the case where we're NOT relaying the write.